packages/siwe/lib/utils.ts

```diff
@@ -49,7 +50,32 @@ export const generateNonce = (): string => {
   return nonce;
 };
 
-export const checkInvalidKeys = <T>(obj: T, keys: Array<keyof T>) : Array<keyof T> => {
+/**
+ * This method matches the given date string against the ISO-8601 regex and also
+ * performs checks if it's a valid date.
+ * @param inputDate any string to be validated against ISO-8601
+ * @returns boolean indicating if the providade date is valid and conformant to ISO-8601
+ */
+export const isValidISO8601Date = (inputDate: string): boolean => {
+  /* Split groups and make sure inputDate is in ISO8601 format */
+  const inputMatch = ISO8601.exec(inputDate);
+
+  /* if inputMatch is null the date is not ISO-8601 */
+  if (!inputDate) {
+    return false;
+  }
+
+  /* Creates a date object with input date to parse for invalid days e.g. Feb, 30 -> Mar, 01 */
+  const inputDateParsed = new Date(inputMatch.groups.date).toISOString();
```

Comment:
Can it parse a time like `25:00:00`? In which case this new date would need to be built with the whole datetime

Reply:
No, it will never be a time like 25:00:00 the regexp makes sure of that.

Reply:
right, indeed